### PR TITLE
Adds Text Inputs and Buttons Selector Shorthand

### PR DESCRIPTION
**Pull request by bhough**
Adds shorthand helper for textInput  and buttonsselectors, including active, focus, and hover states.

### CODE REVIEW

**Comment by bhough** [comment]
After working on the Buttons shorthand, I'm going to refactor this a bit to use a shared helper and leverage tagged template literals to produce the selectors.

File: src/internalHelpers/_statefulSelectors.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  const stateSuffix = state ? `:${state}` : ''
+  return template(stateSuffix)
+}
+
+type State =
+  | typeof(undefined)
+  | null
+  | string;
+
+function statefulSelectors(states: Array<State>, template: Function, stateMap: ?Array<State>) {
```
Comment (mxstbr) [inline src/internalHelpers/_statefulSelectors.js]: Would be nice to have a small high-level comment explaining why we need this!

File: src/shorthands/buttons.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ *   [buttons('active')]: {
+ *     'border': 'none'
+ *   }
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   ${buttons('active')} {
+ *     border: none;
+ *   }
```
Comment (mxstbr) [inline src/shorthands/buttons.js]: This actually doesn't make sense like this in `styled-components`, if anything this should be:

```JS
const div = styled.div`
  > ${buttons('active')} {
    border: none;
  }
`
```

File: src/shorthands/textInputs.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * // Styles as object usage
+ * const styles = {
+ *   [textInputs('active')]: {
+ *     'border': 'none'
+ *   }
+ * }
+ *
+ * // styled-components usage
+ * const div = styled.div`
+ *   ${textInputs('active')} {
```
Comment (mxstbr) [inline src/shorthands/textInputs.js]: Same as above, needs `>`

**Comment by bhough** [comment]
@mxstbr should be good now assuming CI passes.

File: .documentation.json [hunk 4]
```diff
@@ -10,16 +10,29 @@
     },
     "clearFix",
     "ellipsis",
-    "hiDPI",
     "hideText",
+    "hiDPI",
+    "selection",
+    "placeholder",
+    "retinaImage",
```
Comment (mxstbr) [inline .documentation.json]: Those were done in #59 🙈  Would you mind merging that first, rebasing on top of master and then adding the shorthands? Thanks and sorry for the troubles!

**Comment by bhough** [comment]
Ok @mxstbr rebased on master post #59 merge.

**Pull request merged**